src/backend/columnar/cstore_metadata_tables.c

```diff
@@ -1087,7 +1087,11 @@ DatumToBytea(Datum value, Form_pg_attribute attrForm)
 	{
 		if (attrForm->attbyval)
 		{
-			store_att_byval(VARDATA(result), value, attrForm->attlen);
+			Datum tmp;
+			store_att_byval(&tmp, value, attrForm->attlen);
+
+			memcpy_s(VARDATA(result), datumLength + VARHDRSZ,
+					 &tmp, attrForm->attlen);
```

Comment:
should we do something like ?:
```c
ptr += info[dim].typlen;
```


like in https://github.com/postgres/postgres/blob/ca3b37487be333a1d241dab1bbdd17a211a88f43/src/backend/statistics/mcv.c#L873-L886

Reply:
Also, still seeing the same warning on https://github.com/citusdata/citus-enterprise/commits/release-10.0-Hadi-fix

```c
/10.1.0.citus~pre.20210217.9062783/src/backend/distributed/../columnar/cstore_metadata_tables.c:1091:4: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
    store_att_byval(&tmp, value, attrForm->attlen);
```

Reply:
In the postgress code it is unpacking multiple values so it needs to advance the pointer. Here we are unpacking only one value.